signals/webhook/webhook.go

```diff
@@ -38,27 +37,67 @@ const (
 
 // Note: micro requires stateless operation so the Listen() method should not use the
 // receive struct to save or modify state.
-type webhook struct{}
+// webhooks are special in that they should only have one http Server since the port is fixed at runtime
+// this means that webhook signals are stateful, however losing this state is not a concern since
+// the connections will be re-initialized by the SignalClient
+type webhook struct {
+	srv *http.Server
+	sync.RWMutex
+	inactiveEndpoints map[string]struct{}
+}
+
+// New creates a new webhook listener for the specified port
+func New(port int) sdk.Listener {
+	srv := &http.Server{
```

Comment:
Question about webhook signal in-general. 

The sensor yaml for webhooks will have the following:
```
    - name: webhook
      webhook:
        endpoint: /app
        method: POST
```

As a user, how can one reach the `/app` endpoint? Is there a well-defined Kubernetes service which can be used for accessing this from within the cluster?

Reply:
the `hack/k8s/manifests/services/webhook.yaml` has a deployment and a load balancer service for this signal. does this answer your question?